exchange.lisp

```diff
@@ -289,16 +289,14 @@ need-to-use basis, rather than upon initial loading of the exchange API.")
   (:method :around (gate pubkey secret request)
     (rplacd request (call-next-method gate pubkey secret (cdr request)))))
 
-(defmethod perform ((gate gate))
-  (with-slots (input output . #1=(exchange pubkey secret cache)) gate ;¡ has-a !
-    (when (send-blocks-p output) (setf cache (recv input)))
-    (send (slot-value gate 'output) (gate-post . #1#))))
+(defmethod perform recv input ((gate gate))
+  (with-slots (output . #1=(exchange pubkey secret input)) gate ; ¡ has-a !
+    (send output (gate-post . #1#))))
 
```

Comment:
This code is cleaner, as the select method combination handled the synchronisation.